src/pipeline/src/manager/table.rs

```diff
@@ -0,0 +1,450 @@
+                rows: vec![Row {
+                    values: vec![
+                        ValueData::StringValue(name.to_string()).into(),
+                        ValueData::StringValue(schema.to_string()).into(),
+                        ValueData::StringValue(content_type.to_string()).into(),
+                        ValueData::StringValue(pipeline.to_string()).into(),
+                        ValueData::TimestampNanosecondValue(now.value()).into(),
+                    ],
+                }],
+            }),
+        };
+
+        let requests = RowInsertRequests {
+            inserts: vec![insert],
+        };
+
+        let output = self
+            .inserter
+            .handle_row_inserts(
+                requests,
+                Self::query_ctx(&table_info),
+                &self.statement_executor,
+            )
+            .await
+            .context(InsertPipelineSnafu)?;
+
+        info!(
+            "Inserted pipeline: {} into {} table: {}, output: {:?}.",
+            name,
+            PIPELINE_TABLE_NAME,
+            table_info.full_table_name(),
+            output
+        );
+
+        Ok(now)
+    }
+
+    /// Get a pipeline by name.
+    /// If the pipeline is not in the cache, it will be get from table and compiled and inserted into the cache.
+    pub async fn get_pipeline(
+        &self,
+        schema: &str,
+        name: &str,
+        version: Option<String>,
+    ) -> Result<Arc<Pipeline<GreptimeTransformer>>> {
+        if let Some(pipeline) =
+            self.get_compiled_pipeline_from_cache(schema, name, version.as_deref())
+        {
+            return Ok(pipeline);
+        }
+
+        let pipeline = self
+            .find_pipeline_by_name(schema, name, version.as_ref())
+            .await?;
+        let compiled_pipeline = Arc::new(Self::compile_pipeline(&pipeline.0)?);
+
+        self.pipelines.insert(
+            Self::generate_pipeline_cache_key(schema, name, version.as_deref()),
+            compiled_pipeline.clone(),
+        );
+        Ok(compiled_pipeline)
+    }
+
+    /// Insert a pipeline into the pipeline table and compile it.
+    /// The compiled pipeline will be inserted into the cache.
+    pub async fn insert_and_compile(
+        &self,
+        schema: &str,
+        name: &str,
+        content_type: &str,
+        pipeline: &str,
+    ) -> Result<Arc<Pipeline<GreptimeTransformer>>> {
+        let compiled_pipeline = Arc::new(Self::compile_pipeline(pipeline)?);
+        // we will use the version in the future
+        let version = self
+            .insert_pipeline_to_pipeline_table(schema, name, content_type, pipeline)
+            .await?;
+
+        {
+            self.pipelines.insert(
+                Self::generate_pipeline_cache_key(schema, name, None),
+                compiled_pipeline.clone(),
+            );
+            self.pipelines.insert(
+                Self::generate_pipeline_cache_key(schema, name, Some(&version.to_iso8601_string())),
```

Comment:
Why not use `version.value()`? It returns the timestamp value and we can format it to string later.